core/domain/topic_domain.py

```diff
@@ -871,6 +830,57 @@ def __init__(
         self.topic_model_created_on = topic_model_created_on
         self.topic_model_last_updated = topic_model_last_updated
 
+    def validate(self):
+        """Validates all properties of this topic summary.
+
+        Raises:
+            ValidationError: One or more attributes of the Topic summary
+                are not valid.
+        """
+        if not isinstance(self.name, basestring):
+            raise utils.ValidationError('Name should be a string.')
+        if self.name == '':
+            raise utils.ValidationError('Name field should not be empty')
+
+        if not isinstance(self.canonical_name, basestring):
+            raise utils.ValidationError('Canonical name should be a string.')
+        if self.canonical_name == '':
+            raise utils.ValidationError(
+                'Canonical name field should not be empty')
+
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Expected language code to be a string, received %s' %
+                self.language_code)
+        if not utils.is_valid_language_code(self.language_code):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+
+        if not isinstance(self.canonical_story_count, int):
+            raise utils.ValidationError(
+                'Expected canonical story count to be a integer, '
+                'received \'%s\'' % self.canonical_story_count)
+
+        if not isinstance(self.additional_story_count, int):
+            raise utils.ValidationError(
+                'Expected additional story count to be a integer, '
+                'received \'%s\'' % self.additional_story_count)
+
+        if not isinstance(self.uncategorized_skill_count, int):
+            raise utils.ValidationError(
+                'Expected uncategorized skill count to be a integer, '
+                'received \'%s\'' % self.uncategorized_skill_count)
+
+        if not isinstance(self.total_skill_count, int):
```

Comment:
Should this be >= uncategorized_skill_count?

Reply:
Yes, done.